policies/templates/gcp_compute_network_interface_whitelist_v1.yaml

```diff
@@ -0,0 +1,86 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Checks network interfaces on a compute instance against a provided whitelist of networks
+
+apiVersion: templates.gatekeeper.sh/v1alpha1
+kind: ConstraintTemplate
+metadata:
+  name: gcp-compute-network-interface-whitelist-v1
+spec:
+  crd:
+    spec:
+      names:
+        kind: GCPComputeNetworkInterfaceWhitelistConstraintV1
+        plural: gcpcomputenetworkinterfacewhitelistconstraintsV1
+      validation:
+        openAPIV3Schema:
+            whitelist:
+              description: "A list of full object URIs of the networks allowed, for example: ['https://www.googleapis.com/compute/v1/projects/vpc-sc-pub-sub-billing-alerts/global/networks/default1']"
+              type: array
+              items: string
+  targets:
+    validation.gcp.forsetisecurity.org:
+      rego: | #INLINE("validator/compute_network_interface_whitelist.rego")
+            #
+            # Copyright 2019 Google LLC
+            #
+            # Licensed under the Apache License, Version 2.0 (the "License");
+            # you may not use this file except in compliance with the License.
+            # You may obtain a copy of the License at
+            #
+            #      http://www.apache.org/licenses/LICENSE-2.0
+            #
+            # Unless required by applicable law or agreed to in writing, software
+            # distributed under the License is distributed on an "AS IS" BASIS,
+            # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+            # See the License for the specific language governing permissions and
+            # limitations under the License.
+            #
+            
+            package templates.gcp.GCPComputeNetworkInterfaceWhitelistConstraintV1
+            
+            import data.validator.gcp.lib as lib
+            
+            deny[{
+            	"msg": message,
+            	"details": metadata,
+            }] {
+            	constraint := input.constraint
+            	asset := input.asset
+            	asset.asset_type == "compute.googleapis.com/Instance"
+            
+            	lib.get_constraint_params(input.constraint, params)
+            
+            	instance := asset.resource.data
+            
+            	interfaces := lib.get_default(instance, "networkInterface", [])
+            	interface := interfaces[_]
+            	full_network_uri := interface.network
+            
+            	whitelist := lib.get_default(params, "whitelist", [])
+            	allowed_networks := {n | n = whitelist[_]}
+            
+            	access_configs := lib.get_default(interface, "accessConfig", [])
+            
+            	is_external_network := count(access_configs) > 0
+            	is_network_whitelisted := count({full_network_uri} - allowed_networks) == 0
+            
+            	is_external_network == true
```

Comment:
Is there a reason to use a temporary object here? Why not just the test statements directly here (and test for != 0 for L78)?

Reply:
I think creating an `is_external_network` var helps readability.

Reply:
fair point :)